uxarray/grid/neighbors.py

```diff
@@ -0,0 +1,220 @@
+import numpy as np
+from numpy import deg2rad
+import warnings
+
+from sklearn.neighbors import BallTree as SKBallTree
+from typing import Optional, Union
+
+from uxarray.constants import INT_DTYPE
+
+
+class BallTree:
+    """Custom BallTree datastructure written around the
+    ``sklearn.neighbors.BallTree`` implementation for use with either the
+    corner (``Mesh2_node_x``, ``Mesh2_node_y``) or center (``Mesh2_face_x``,
+    ``Mesh2_face_y``) nodes of the inputted unstructured grid.
+
+    Notes
+    -----
+    See `sklearn.neighbors.BallTree <https://scikit-learn.org/stable/modules/generated/sklearn.neighbors.BallTree.html>`__
+    for further information about the wrapped data structure.
+    """
+
+    def __init__(self, grid, node_type='corner'):
+
+        # construct tree from corner nodes
```

Comment:
What are the different use cases for this? Why would we want a tree from corner nodes vs center nodes?

Reply:
Both are needed depending on where our data is defined (i.e. on each node or face center)

### Corner Nodes
Used for computing the distance between the corners that make up each face. 

### Center Nodes
Used for computing the distance between the centers of each face